### PR TITLE
BatchedMesh: Add "toJSON" and "copy" support

### PR DESCRIPTION
**Pull request by gkjohnson**
Related issue: #22376, #27111 

**Description**

Add support for "toJSON" and "copy" to "BatchedMesh".

**Upcoming PRs**
- Raycasting, sorting support
- Object-level bounds
- Add support to ObjectLoader
- Move to core
- Documentation

### CODE REVIEW

File: src/core/Object3D.js [hunk 1]
```diff
@@ -719,6 +719,26 @@ class Object3D extends EventDispatcher {
...
+			object._active = this._active;
+
+			object._maxGeometryCount = this._maxGeometryCount;
+			object._maxVertexCount = this._maxVertexCount;
+			object._maxIndexCount = this._maxIndexCount;
+
+			object._geometryInitialized = this._geometryInitialized;
+			object._geometryCount = this._geometryCount;
+
+			object._matricesTexture = this._matricesTexture.toJSON();
```
Comment (Mugen87) [inline src/core/Object3D.js:738]: Can we avoid serializing the internal texture? Ideally, it is just recreated in `BatchedMesh` if it is missing.

Besides, can we avoid the underscore syntax in the JSON object? 

Also, at some point this logic should be moved to `BatchedMesh.toJSON()`. `Object3D` should access private variables of other subclasses.

Comment (gkjohnson) [reply src/core/Object3D.js:738]: > Can we avoid serializing the internal texture? Ideally, it is just recreated in BatchedMesh if it is missing.

This texture is how the matrices are canonically stored. I can generate a list of serialized matrices from the texture instead, though, if that's acceptable.

> Besides, can we avoid the underscore syntax in the JSON object?

Yup! I'll fix it.

> Also, at some point this logic should be moved to BatchedMesh.toJSON(). Object3D should not access private variables of subclasses.

I thought so, too, but this looked like how all the other classes like InstancedMesh were working. Though it's unclear to me why this is needed.

Comment (Mugen87) [reply src/core/Object3D.js:738]: > This texture is how the matrices are canonically stored. I can generate a list of serialized matrices from the texture instead, though, if that's acceptable.

I forgot after #27130 there is only the texture that holds the matrices. In this case, I'm fine with serializing directly. the texture

**Review by Mugen87 — APPROVED** [review]

File: src/core/Object3D.js (hunk 1 above)
Comment (Mugen87) [reply src/core/Object3D.js:738]: BTW: `ObjectLoader` should also be able to handle `BatchedMesh`. Maybe we can add this to the todo list.

**Comment by Mugen87** [comment]
Sorry, approved a bit too early. The removal of the underscore in the JSON would still be nice^^!

**Comment by gkjohnson** [comment]
> Sorry, approved a bit too early. The removal of the underscore in the JSON would still be nice^^!

Fixed!

> BTW: ObjectLoader should also be able to handle BatchedMesh. Maybe we can add this to the todo list.

Added to the list!

File: examples/jsm/objects/BatchedMesh.js [hunk 2]
```diff
@@ -563,17 +563,29 @@ class BatchedMesh extends Mesh {
 
 	}
 
-	copy() {
+	copy( source ) {
```
Comment (Methuselah96) [inline examples/jsm/objects/BatchedMesh.js:566]: Can we have `copy` return `this` to make it consistent with parent classes `Object3D` and `Mesh`?

**Pull request merged**